charts/rancher-vsphere-csi/templates/node/daemonset.yaml

```
@@ -117,7 +117,15 @@ spec:
             - name: CSI_ENDPOINT
               value: unix:///csi/csi.sock
             - name: MAX_VOLUMES_PER_NODE
-              value: "59" # Maximum number of volumes that controller can publish to the node. If value is not set or zero Kubernetes decide how many volumes can be published by the controller to the node.
+              {{- if .Values.maxPvscsiTargetsPerVm.enabled }}
+              {{-   if gt .Values.csiNode.maxVolumesPerNode 255 }}
```

This is a string. You can't call `gt` on a string.
```suggestion
              {{-   if gt (int .Values.csiNode.maxVolumesPerNode) 255 }}
```

Since we're doing range checks here, do you also want to ensure that it's a valid int, and greater than 0 or some other lower bound?